File: javascript/programmers/javascript/level2/큰수만들기.js
const number = "4177252841";
const k = 4;

const solution = (number, k) => {
  const stack = [];
  for (let i = 0; i < number.length; i++) {
    let now = number[i];
    while (k > 0 && stack[stack.length - 1] < now) {
      stack.pop();
      k--;
    }
    stack.push(now);
  }
  stack.splice(stack.length - k, k);
  let answer = stack.join("");

  return answer;
};

console.log(solution(number, k));

File: python/programmers/level2/큰 수 만들기.py
def solution(number, k):
    answer = number
    i = 0
    while i < len(number) - 1 and k > 0:
        if answer[i] < answer[i + 1]:
            answer = answer[:i] + answer[i + 1:]
            k -= 1
            if i != 0:
                i -= 1
        else:
            i += 1
    if k > 0:
        return answer[:-k]
    return answer


print(solution("4177252841", 4))
